### PR TITLE
gh-124043: Disallow mixing `--with-trace-refs` and `--disable-gil`

### PR DESCRIPTION
**Pull request by ZeroIntensity**
cc @colesbury

* Issue: gh-124043

### CODE REVIEW

**Comment by ZeroIntensity** [comment]
Marking as draft while I fix CI.

File: Misc/NEWS.d/next/Build/2024-09-13-17-48-37.gh-issue-124043.Bruxpq.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Building using ``--with-trace-refs`` is (temporarily) disallowed when the
```
Comment (Eclips4) [inline Misc/NEWS.d/next/Build/2024-09-13-17-48-37.gh-issue-124043.Bruxpq.rst]:
```suggestion
Building using :option:`--with-trace-refs` is (temporarily) disallowed when the
```

**Comment by Eclips4** [comment]
LGTM. I'd prefer to backport it to 3.13 branch (we can wait until 3.13.1, this is not a big deal, IMO)

**Review by colesbury — APPROVED** [review]

**Comment by colesbury** [comment]
The PR is pretty small and straightforward, but If @erlend-aasland or @corona10 could take a look at this too, I'd appreciate it.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**